File: app/src/main/java/com/grsu/guideapp/adapters/AboutAdapter.java
package com.grsu.guideapp.adapters;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.grsu.guideapp.R;
import com.grsu.guideapp.base.listeners.ItemClickListener;
import com.grsu.guideapp.fragments.about.AboutItem;
import com.grsu.guideapp.holders.about.AboutHolder;
import java.util.List;

public class AboutAdapter extends RecyclerView.Adapter<AboutHolder> {

    private List<AboutItem> aboutItems;
    private ItemClickListener listener;

    public AboutAdapter(List<AboutItem> aboutItems, ItemClickListener listener) {
        this.aboutItems = aboutItems;
        this.listener = listener;
    }

    @NonNull
    @Override
    public AboutHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater from = LayoutInflater.from(parent.getContext());
        View view = from.inflate(R.layout.item_about, parent, false);
        return new AboutHolder(view, listener);
    }

    @Override
    public void onBindViewHolder(@NonNull AboutHolder holder, int position) {
        AboutItem item = aboutItems.get(position);
        holder.bind(item);
    }

    @Override
    public int getItemCount() {
        return aboutItems != null ? aboutItems.size() : 0;
    }
}

File: app/src/main/java/com/grsu/guideapp/ui/adapters/AboutAdapter.kt
package com.grsu.guideapp.ui.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.grsu.guideapp.R
import com.grsu.guideapp.ui.fragments.about.AboutItem
import com.grsu.guideapp.ui.holders.about.AboutHolder

class AboutAdapter(
    private var aboutItems: List<AboutItem>,
    private val listener: (View, Int) -> Unit
) : RecyclerView.Adapter<AboutHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AboutHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_about, parent, false)
        return AboutHolder(view, listener)
    }

    override fun onBindViewHolder(holder: AboutHolder, position: Int) = holder.bind(aboutItems[position])

    override fun getItemCount(): Int = aboutItems.size
}